types/three/three-core.d.ts

```diff
@@ -6119,10 +6119,14 @@ export namespace SceneUtils {
 }
 
 export namespace ShapeUtils {
-    export function area(contour: number[]): number;
-    export function triangulate(contour: number[], indices: boolean): number[];
-    export function triangulateShape(contour: number[], holes: any[]): number[];
-    export function isClockWise(pts: number[]): boolean;
+    export function area(contour: ArrayLike<{x: number, y: number}>): number;
+
+    export function isClockWise(pts: ArrayLike<{x: number, y: number}>): boolean;
+
+    export function triangulate<T extends ArrayLike<{x: number, y: number}> | number[][]>(contour: T, indices?: boolean): T[];
+    export function triangulate(contour: ArrayLike<{x: number, y: number}> | number[][], indices: true): number[][];
```

Comment:
Good choice on using generics when the return type depends on the input type. I have a few points to make:
1. We should have an overload for `true`, `false`, `boolean` (eg. not statically known) and no `indices` param, as they all have different signatures. And a test for each one.
2. I think that the `T` should not be for the entire `ArrayLike<...>`, instead `T extends {x: number, y: number}` and the function takes `ArrayLike<T>`, this is mainly for correctness of the return type, because the function does not return the same `ArrayLike` type that was passed in, it will return regular JS Arrays. Specifically this: `[T,T,T][]`.
3. I can't see how the function validly accepts `number[][]`, I see it doesn't throw anything in your jsfiddle but i just can't see the implementation handling this in a way that makes sense. Can you help me, am I missing something?
4. In the `indices=true` case I think `[number, number, number][]` is an even better return type (similar to `T,T,T` in point 2). I think the same applies to `triangulateShape`.